lib/action_view/component/base.rb

```diff
@@ -129,15 +129,11 @@ def call_method_name(variant)
 
         def source_location
           @source_location ||=
-              begin
-                # Require #initialize to be defined so that we can use
-                # method#source_location to look up the file name
-                # of the component.
-                #
-                # If we were able to only support Ruby 2.7+,
-                # We could just use Module#const_source_location,
-                # rendering this unnecessary.
-                #
+              if const_source_location_supported?
+                const_source_location(self.name)[0]
+              else
+                # For Ruby < 2.7.0, use `method#source_location` on `#initialize`
```

Comment:
```suggestion
                # Require #initialize to be defined so that we can use	                const_source_location(self.name)[0]
                # method#source_location to look up the file name
                # of the component
```

Reply:
done.